File: Nemes Tihamér/Sort/sort.py
class Rec(object):
    def __init__(self, name, age):
        self.name = name
        self.age = age
    def __str__(self):
        return self.name + " " + self.age
    def __lt__(self, other):
        return self.age < other.age

def kiir(adatok):
    for r in adatok:
        print (r)

def main():
    #n = input() Python 2
    n = int(input())
    adatok = []
    for i in range(0,n):
        #s = raw_input() Python 2
        s = input()
        temp = s.split()
        rec = Rec(temp[0],temp[1])
        adatok.append(rec)

    kiir(adatok)
    print()

    adatok.sort()
    #adatok = sorted(adatok, reverse = True)

    kiir(adatok)
    
main()

File: Nemes Tihamér/Sort/Sort.cs
using System;
using System.Collections.Generic;

public class Rec : IComparable<Rec>
{
	public string name;
	public int age;
	
	public Rec(string name, int age)
	{
		this.name = name;
		this.age = age;
	}

	public override string ToString()
	{
		return this.name + " " + this.age;
	}
	
	public int CompareTo(Rec o)
	{
		if(this.age < o.age)
		{
			return -1;
		}
		else if(this.age == o.age)
		{
			return 0;
		}
		else
		{
			return 1;
		}
	}
}
	
class Sort
{
	public static void kiir(List<Rec> adatok)
	{
		foreach(Rec r in adatok)
		{
			Console.WriteLine(r);
		}
	}
	
	public static void Main()
	{
		int n = Convert.ToInt32(Console.ReadLine());
		List<Rec> adatok = new List<Rec>();
		for(int i = 0; i < n; i++)
		{
			string[] temp = Console.ReadLine().Split();
			string name = temp[0];
			int age = Convert.ToInt32(temp[1]);
			adatok.Add(new Rec(name, age));
		}
		
		kiir(adatok);
		Console.WriteLine();
		
		adatok.Sort();
		//adatok.Reverse();
		
		kiir(adatok);
		
	}
}
